games-arcade/defendguin/defendguin-0.0.12-r1.ebuild

```diff
@@ -1,40 +1,42 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-inherit desktop
+EAPI=8
 
-DESCRIPTION="A clone of the arcade game Defender, but with a Linux theme"
+inherit desktop toolchain-funcs
+
+DESCRIPTION="Clone of the arcade game Defender, but with a Linux theme"
 HOMEPAGE="http://www.newbreedsoftware.com/defendguin/"
-SRC_URI="ftp://ftp.tuxpaint.org/unix/x/${PN}/src/${P}.tar.gz"
+SRC_URI="ftp://ftp.tuxpaint.org/unix/x/defendguin/src/${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-IUSE=""
 
-DEPEND="media-libs/sdl-mixer[mod]
-	media-libs/libsdl[video]"
+DEPEND="
+	media-libs/libsdl[joystick,sound,video]
+	media-libs/sdl-mixer[mod]"
 RDEPEND="${DEPEND}"
 
 src_prepare() {
 	default
+	tc-export CC
 	sed -i \
-		-e "s:\$(DATA_PREFIX):/usr/share/${PN}/:" \
+		-e "s|\$(DATA_PREFIX)|${EPREFIX}/usr/share/${PN}/|" \
 		-e '/^CFLAGS=.*-O2/d' \
-		-e '/^CFLAGS=/s:=:+= $(LDFLAGS) :' \
-		Makefile \
-		|| die
-	rm -f data/images/*.sh
+		-e '/^CFLAGS=/s|=|+= $(LDFLAGS) |' \
+		Makefile || die
+	rm data/images/l2r.sh || die
 }
 
 src_install() {
 	dobin ${PN}
+
 	insinto /usr/share/${PN}
 	doins -r ./data/*
 
 	newicon data/images/ufo/ufo0.bmp ${PN}.bmp
```

Comment:
I thought too that `bmp` isn't a valid xdg icon, but since it worked i kept it. Anyway, i'll prefer using `png` too so thanks for uploading the icon :)

Reply:
Yeah, it'll depend on what you use. Some app launchers won't care while others only follow the specification making jpg, gif, and bmp all bad.

https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#definitions